client/www/scripts/modules/licenses/licenses.controllers.js

```diff
@@ -0,0 +1,75 @@
+Licenses.controller('LicensesMainController', [
+  '$scope',
+  '$q',
+  '$log',
+  'LicensesService',
+  function ($scope, $q, $log, LicensesService) {
+    $scope.errors = {
+      renewal: {},
+      generic: []
+    };
+
+    $scope.hasErrors = false;
+
+    window.setScrollView('.common-instance-view-container');
+
+    function handleSuccess(results){
+
+      //handle calls that failed
+      results.forEach(function(res){
+        if ( res.data && res.data.error ) {
+          handleError(res);
+        }
+      });
+
+      //refresh the list
+      LicensesService.getProductsAndLicenses()
+        .then(function(products){
+          $scope.allProducts = products;
+        });
+    }
+
```

Comment:
I think you can simplify a lot of this controller by moving a bunch of logic into services:

$scope.allProducts = LicenseService.getProductsAndLicenses()
     .then(function(response){
         retrurn response;
     })
     .catch(function(error){
     ...
    });
I'm probably missing something  let me know if you want to review together